client/checkout/classic/new-upe.js

```diff
@@ -0,0 +1,242 @@
+	 * It is called when the page is loaded.
+	 *
+	 */
+	$( document.body ).on( 'updated_checkout', () => {
+		$( '.wcpay-upe-element' )
+			.toArray()
+			.forEach( ( domElement ) => {
+				const paymentMethodType = $( domElement ).data(
+					'payment-method-type'
+				);
+				const upeElement =
+					gatewayUPEComponents[ paymentMethodType ].upeElement;
+				if ( upeElement ) {
+					upeElement.mount( domElement );
+				} else {
+					createAndMountPaymentElement(
+						paymentMethodType,
+						domElement
+					);
+				}
+			} );
+	} );
+
+	/**
+	 * Block UI to indicate processing and avoid duplicate submission.
+	 *
+	 * @param {Object} $form The jQuery object for the form.
+	 */
+	const blockUI = ( $form ) => {
+		$form.addClass( 'processing' ).block( {
+			message: null,
+			overlayCSS: {
+				background: '#fff',
+				opacity: 0.6,
+			},
+		} );
+	};
+
+	/**
+	 * Creates the UPE element and mounts it to the DOM element.
+	 * No Stripe PaymentIntent is created at this point.
+	 *
+	 * @param {string} paymentMethodType The type of payment method, e.g. 'card', 'giropay', etc.
+	 * @param {string} domElement The selector of the DOM element of particular payment method to mount the UPE element to.
+	 */
+	async function createAndMountPaymentElement(
+		paymentMethodType,
+		domElement
+	) {
+		const options = {
+			mode: 'payment',
+			currency: getUPEConfig( 'currency' ).toLowerCase(),
+			amount: Number( getUPEConfig( 'cartTotal' ) ),
+			paymentMethodCreation: 'manual',
+			paymentMethodTypes: [ paymentMethodType ],
+			appearance: appearance,
+		};
+
+		if ( ! fingerprint ) {
+			try {
+				const { visitorId } = await getFingerprint();
+				fingerprint = visitorId;
+			} catch ( error ) {
+				// Do not mount element if fingerprinting is not available
+				showErrorCheckout( error.message );
+
+				return;
+			}
+		}
+
+		const elements = api.getStripe().elements( options );
+		gatewayUPEComponents[ paymentMethodType ].elements = elements;
+
+		const upeElement = elements.create( 'payment', {
+			...getUpeSettings(),
+			wallets: {
+				applePay: 'never',
+				googlePay: 'never',
+			},
+		} );
+
+		upeElement.mount( domElement );
+		gatewayUPEComponents[ paymentMethodType ].upeElement = upeElement;
+	}
+
+	function getUpeSettings() {
+		const upeSettings = {
+			fields: {
+				billingDetails: getHiddenBillingFields(
+					getUPEConfig( 'enabledBillingFields' )
+				),
+			},
+		};
+
+		if ( getUPEConfig( 'cartContainsSubscription' ) ) {
+			upeSettings.terms = getTerms( paymentMethods, 'always' );
+		}
+
+		return upeSettings;
+	}
+
+	const handleCheckout = async ( $form, paymentMethodType ) => {
+		blockUI( $form );
+		const elements = gatewayUPEComponents[ paymentMethodType ].elements;
+		elements.submit().then( ( result ) => {
+			if ( result.error ) {
+				showErrorCheckout( result.error.message );
+			}
+		} );
+		const pm = await api.getStripe().createPaymentMethod( {
```

Comment:
```suggestion
		const paymentMethod = await api.getStripe().createPaymentMethod( {
```
This is pretty pedantic and isn't really something that I'm particularly bothered about, but I think [using full-names for variables is more accepted form](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#naming-conventions).

Reply:
Pedantic but crucial - thanks for it! As part of https://github.com/Automattic/woocommerce-payments/pull/5869/commits/1d22a4c02b72815410512861eb7db60f7eda279e, we now use the fluent interface and the name has disappeared.